ballista/executor/src/circuit_breaker/client.rs

```diff
@@ -0,0 +1,290 @@
+                        scheduler_ids
+                            .insert(registration.task_id, registration.scheduler_id);
+                    }
+
+                    ClientUpdate::SchedulerDeregistration(unregistration) => {
+                        scheduler_unregistrations.push(unregistration);
+                    }
+                }
+            }
+
+            let mut updates_per_scheduler = HashMap::new();
+            let mut seen_keys = HashSet::new();
+
+            for update in updates.into_iter().rev() {
+                // Per request only one update per task is sent
+                // This is why we go from newest to oldest
+                if seen_keys.insert(update.key.clone()) {
+                    let scheduler_id: &String = match scheduler_ids
+                        .get(&update.key.task_id)
+                    {
+                        Some(scheduler_id) => scheduler_id,
+                        None => {
+                            warn!("No scheduler found for task {}", update.key.task_id);
+                            continue;
+                        }
+                    };
+
+                    updates_per_scheduler
+                        .entry(scheduler_id.clone())
+                        .or_insert_with(Vec::new)
+                        .push(update);
+                }
+            }
+
+            for (scheduler_id, updates) in updates_per_scheduler {
+                let mut request_updates = Vec::with_capacity(updates.len());
+
+                for update in updates {
+                    let key = update.key.into();
+
+                    request_updates.push(protobuf::CircuitBreakerUpdate {
+                        key: Some(key),
+                        percent: update.percent,
+                    })
+                }
+
+                let mut scheduler = match get_scheduler
+                    .get_or_create_scheduler_client(&scheduler_id)
+                    .await
+                {
+                    Ok(scheduler) => scheduler,
+                    Err(e) => {
+                        warn!("Failed to get scheduler {}: {}", scheduler_id, e);
+                        continue;
+                    }
+                };
+
+                let request = CircuitBreakerUpdateRequest {
+                    updates: request_updates,
+                };
+
+                match scheduler.send_circuit_breaker_update(request).await {
+                    Err(e) => warn!(
+                        "Failed to send circuit breaker update to scheduler {}: {}",
+                        scheduler_id, e
+                    ),
+                    Ok(response) => {
+                        let commands = response.into_inner().commands;
+
+                        for command in commands {
+                            if let Some(key_proto) = command.key {
+                                let key = key_proto.into();
+
+                                if let Some(state) = state_per_task.get(&key) {
```

Comment:
Similar to above can use `Ordering::Release` here since we are only doing a store